File: training/python/de/sots/cellarsandcentaurs/domain/service/creature_service.py
from __future__ import annotations

import logging
from typing import TYPE_CHECKING

from ..model import creature
from .creatures import Creatures

if TYPE_CHECKING:
    from ..model.creature_id import CreatureId

logger = logging.getLogger(__name__)


class CreatureService:
    def __init__(self, creatures: Creatures):
        self.creatures = creatures

    def save(self, creature_to_save: creature.Creature) -> None:
        logger.info("saving creature")
        self.creatures.save(creature_to_save)

    def find(self, creature_id: CreatureId) -> creature.Creature:
        return self.creatures.find(creature_id)

File: training/javascript/src/de/sots/cellarsandcentaurs/domain/service/CreatureService.js
import { createLogger } from "winston";
import { Creatures } from "./Creatures.js";
import { Creature } from "../model/Creature.js";

const logger = createLogger();

export class CreatureService {
    /**
     * @param {Creatures} creatures
     */
    constructor(creatures) {
        this.creatures = creatures;
    }

    /**
     * @param {Creature} creature
     */
    save(creature) {
        if (!(creature instanceof Creature)) {
            throw new TypeError("not a creature");
        }
        logger.info("saving creature");
        this.creatures.save(creature);
    }
}
